computer_vision/pooling_functions.py

```diff
@@ -115,21 +120,52 @@ def avgpooling(arr: np.ndarray, size: int, stride: int) -> np.ndarray:
     return updated_arr
 
 
-# Main Function
+# Global Average pooling Function
+def global_average_pooling(arr: np.ndarray) -> np.ndarray:
+    """
+    This function performs global average pooling on the input 2D matrix (image).
+    Args:
+        arr: numpy array
+            The input 2D matrix.
+    Returns:
+        numpy array of the global average pooled matrix.
+
+    >>> global_average_pooling(np.array([[1,2,3],[5,6,7],[9,10,11],[13,14,15]]))
+    array([[8.]])
+    >>> global_average_pooling(np.array([[147, 180, 122],[241, 76, 32],[126, 13, 157]]))
+    array([[121.55555556]])
+    """
+    arr = np.array(arr)
+
+    # Calculate the average of the entire input matrix
+    average_value = np.mean(arr)
+
+    # Create a matrix with a single element, which is the average value
+    global_avg_pooled_matrix = np.array([[average_value]])
+
+    return global_avg_pooled_matrix
```

Comment:
Please just have your function return the value as a float rather than returning it in a numpy array